static/app/components/badge/groupPriority.tsx

```diff
@@ -85,21 +88,39 @@ function useLastEditedBy({
 
 export function makeGroupPriorityDropdownOptions({
   onChange,
+  hasStreamlinedIssueStream,
 }: {
+  hasStreamlinedIssueStream: boolean;
   onChange: (value: PriorityLevel) => void;
 }) {
   return PRIORITY_OPTIONS.map(priority => ({
     textValue: PRIORITY_KEY_TO_LABEL[priority],
     key: priority,
-    label: <GroupPriorityBadge priority={priority} />,
+    label: (
+      <GroupPriorityBadge
+        variant={hasStreamlinedIssueStream ? 'streamlined' : 'default'}
+        priority={priority}
+      />
+    ),
```

Comment:
Note that I have decided to keep the labels for the badges that are shown in the dropdown, unlike for the badges in the issue stream.